src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,841 @@
+
+        :raises NoImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("stencil_unknown_extent not implemented")
+
+    def stencil_unknown_direction(self, arg, var_accesses=None):
+        '''Not implemented.
+
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NoImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("stencil_unknown_direction not implemented")
+
+    def stencil(self, arg, var_accesses=None):
+        '''Not implemented.
+
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NoImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("stencil not implemented")
+
+    def operator(self, arg, var_accesses=None):
+        '''Create an LFRic operator argument and an ncells argument and add
+        them to the symbol table and argument list. Also declare the
+        associated 'fs_from', 'fs_to' symbols if they have not already
+        been declared so that they can be used to dimension the
+        operator symbol (as well as ncells).
+
+        :param arg: the operator to add.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: if the datatype of the field is \
+            not supported.
+
+        '''
+        # Create and add ndf symbols to the symbol table if they do
+        # not already exist or return the existing ones if they do.
+        fs_from_name = arg.function_space_from.orig_name
+        ndf_symbol_from = self._create_symbol(
+            "ndf_{0}".format(fs_from_name), NumberOfDofsDataSymbol,
+            extra_args=[fs_from_name])
+        fs_to_name = arg.function_space_to.orig_name
+        ndf_symbol_to = self._create_symbol(
+            "ndf_{0}".format(fs_to_name), NumberOfDofsDataSymbol,
+            extra_args=[fs_to_name])
+
+        ncells = NumberOfCellsDataSymbol(
+            "ncell_3d", interface=self._read_access)
+        self._symbol_table.add(ncells)
```

Comment:
Yes that's right. I am just replicating the current API rules. I guess we should only pass ncell_3d once but at the moment the LFRic API specifies that we pass it multiple times. In Fortran you are allowed to pass the same variable more than once so it is not the end of the world. I think passing it once should be a change to the API which should then be reflected here, rather than changing the API explicitely here (which will not change the kernel call generation code and the stub generator.

Reply:
Agreed.